### ISSUE TITLE
v4 <-> InferenceData integration

### ISSUE THREAD

**Issue by OriolAbril** [pymc-devs/pymc#5160]
For now a dump of things to take into account regarding integration of PyMC v4 with InferenceData, which can now be improved signifiicantly further now that the converter is part of the PyMC codebase. 

Goals:
* Converting from MultiTrace to InferenceData does not represent loosing of information -> transformed variables, metadata, values in report...
  * feasible
* Storing sampler arguments, initial values, mass matrix... to allow reproducible sampling given a model object and an inferencedata. A bit related to https://github.com/arviz-devs/arviz/issues/220
  * might not be feasible yet

Related issues:
* https://github.com/arviz-devs/arviz/issues/230
* https://github.com/arviz-devs/arviz/issues/420
* Making the converter Dask compatible 
* https://github.com/arviz-devs/arviz/issues/1224
* https://github.com/arviz-devs/arviz/issues/1257
* https://github.com/arviz-devs/arviz/issues/1509
* https://github.com/arviz-devs/arviz/issues/1748

**Comment by bacalfa** [comment]
Please add a way to return trace information from `pm.sample` (https://discourse.pymc.io/t/inferencedata-is-missing-details-about-divergent-transitions/7740).

**Comment by OriolAbril** [comment]
It might make sense to subclass inferencedata at some point to add extra conventions when reading/writing inferencedata files to netcdf or zarr. For example, it might be possible to retrieve the transformations on posterior variables from the names or attributes of posterior and unconstrained_posterior groups (even if not always). I have also not been following the modelbuilder work, but it might also help a bit there?

**Comment by michaelosthege** [comment]
This is obsolete with the upcoming ArviZ 1.0 release terminating `InferenceData` in favor of `xarray.DataTree`: https://python.arviz.org/en/latest/user_guide/migration_guide.html

**Comment by OriolAbril** [comment]
I agree it needs an update and clean up, but I think only my last comment about subclassing the InferenceData class is affected by the move from InferenceData custom class to xarray provided DataTree class. Both classes are a Python representation of hierarchical array files like netcdf or zarr.

That being said, with the zarr backend and many other changes since v4 and the fact this was mostly a list of other issues it might not be worth to keep this issue open
